parcels/grid.py

```diff
@@ -205,8 +377,8 @@ class GridIndex(object):
     def __init__(self, grid, *args, **kwargs):
         self._cptr = kwargs.pop('cptr', None)
         self.name = grid.name
-        self.xi = 0
-        self.yi = 0
+        self.xi = int(grid.xdim / 2)
+        self.yi = int(grid.ydim / 2)
```

Comment:
is this purely for speedup? To start halfway? Does it matter? If really only for speed, then perhaps wait with this change until we really start benchmarking?

Reply:
Indeed, the final goal of this is to save some time.
But currently, if I remove it, I still have problems with the periodic boundary.
Indeed, I've just discovered the problem exists even without this.
I'm working on it